beacon_node/beacon_chain/src/eth1_chain.rs

```diff
@@ -48,23 +49,31 @@ pub enum Error {
     UnknownPreviousEth1BlockHash,
 }
 
+#[derive(Encode, Decode, Clone)]
+pub struct SszEth1 {
+    use_dummy_backend: bool,
+    backend_bytes: Vec<u8>,
+}
+
 /// Holds an `Eth1ChainBackend` and serves requests from the `BeaconChain`.
-pub struct Eth1Chain<T, E>
+pub struct Eth1Chain<T, E, S>
```

Comment:
Is it possible to make just `from_ssz_container` and `as_ssz_container` generic over `Store` instead of all of the `Eth1Chain`?